### PR TITLE
Return an error on `show_validator`

### PR DESCRIPTION
**Pull request by alexanderbez**
closes: #3314 

----

* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [x] Updated CHANGELOG_PENDING.md

### CODE REVIEW

**Comment by alexanderbez** [comment]
Looks like the `p2p` package is failing but this seems to be an existing regression on the latest `develop`?

**Comment by Liamsi** [comment]
There are a bunch of known non-deterministic test-failures which are known. After a quick search through the open issues, it seems this is a new one: `TestStateLockPOLSafety1`. I'll open an issue.

File: CHANGELOG_PENDING.md [hunk 1]
```diff
@@ -21,5 +21,9 @@ Special thanks to external contributors on this release:
 ### IMPROVEMENTS:
 
 ### BUG FIXES:
-- [consensus] \#3297 Flush WAL on stop to prevent data corruption during
+
+* [consensus] \#3297 Flush WAL on stop to prevent data corruption during
   graceful shutdown
+* [cmd] [\#3314](https://github.com/tendermint/tendermint/issues/3314) Return an
```
Comment (Liamsi) [inline CHANGELOG_PENDING.md]: I think we run a script that adds the links in the end. Not sure if adding a link here directly does cause any problems with that script: https://github.com/tendermint/tendermint/blob/master/scripts/linkify_changelog.py

```suggestion
* [cmd] \#3314 Return an
```

File: cmd/tendermint/commands/show_validator.go [hunk 2]
```diff
@@ -5,18 +5,29 @@ import (
...
+	cmn "github.com/tendermint/tendermint/libs/common"
 	"github.com/tendermint/tendermint/privval"
 )
 
 // ShowValidatorCmd adds capabilities for showing the validator info.
 var ShowValidatorCmd = &cobra.Command{
 	Use:   "show_validator",
 	Short: "Show this node's validator info",
-	Run:   showValidator,
+	RunE:  showValidator,
```
Comment (Liamsi) [inline cmd/tendermint/commands/show_validator.go]: Interesting, this aren't named consistently across commands. Did never notice that.

File: cmd/tendermint/commands/show_validator.go [hunk 3]
```diff
@@ -5,18 +5,29 @@ import (
...
+func showValidator(cmd *cobra.Command, args []string) error {
+	keyFilePath := config.PrivValidatorKeyFile()
+	if !cmn.FileExists(keyFilePath) {
+		return fmt.Errorf("private validator file %s does not exist", keyFilePath)
+	}
+
+	pv := privval.LoadFilePV(keyFilePath, config.PrivValidatorStateFile())
+	bz, err := cdc.MarshalJSON(pv.GetPubKey())
+	if err != nil {
+		return err
```
Comment (Liamsi) [inline cmd/tendermint/commands/show_validator.go]: Do you think we should wrap this with something the user understands? Not sure if whatever amino would return is helpful for the caller.

**Comment by Liamsi** [comment]
After restarting `TestPEXReactorRunning` fails. This one is known: https://github.com/tendermint/tendermint/issues/2176

File: CHANGELOG_PENDING.md (hunk 1 above)
Comment (alexanderbez) [reply CHANGELOG_PENDING.md]: Gotcha. Amended. I saw a mix case usage in the changelog.

**Comment by alexanderbez** [comment]
Interesting @Liamsi -- I'd be happy to help dig into and potentially fix some of the non-deterministic tests.

**Comment by melekes** [comment]
@alexanderbez here is the list https://github.com/tendermint/tendermint/issues?utf8=%E2%9C%93&q=is%3Aopen+is%3Aissue+non-deterministic

File: cmd/tendermint/commands/show_validator.go (hunk 3 above)
Comment (melekes) [reply cmd/tendermint/commands/show_validator.go]: Will do

**Comment by melekes** [comment]
Thanks for contributing to Tendermint 👍

**Pull request merged**